components/service/glean/src/test/java/mozilla/components/service/glean/storages/StringListsStorageEngineTest.kt

```diff
@@ -0,0 +1,258 @@
+        assertEquals(
+            StringListsStorageEngineImplementation.MAX_LIST_LENGTH_VALUE,
+            snapshot2["telemetry.string_list_metric"]?.count()
+        )
+    }
+
+    @Test
+    fun `set() won't allow adding a list longer than the max list length`() {
+        val metric = StringListMetricType(
+            disabled = false,
+            category = "telemetry",
+            lifetime = Lifetime.Ping,
+            name = "string_list_metric",
+            sendInPings = listOf("store1")
+        )
+
+        val stringList: MutableList<String> = mutableListOf()
+        for (i in 1..StringListsStorageEngineImplementation.MAX_LIST_LENGTH_VALUE + 1) {
+            stringList.add("value$i")
+        }
+
+        StringListsStorageEngine.set(metricData = metric, value = stringList)
+
+        // Check that list was truncated.
+        val snapshot = StringListsStorageEngine.getSnapshot(
+            storeName = "store1",
+            clearStore = false)
+        assertEquals(1, snapshot!!.size)
+        assertEquals(true, snapshot.containsKey("telemetry.string_list_metric"))
+        assertEquals(
+            StringListsStorageEngineImplementation.MAX_LIST_LENGTH_VALUE,
+            snapshot["telemetry.string_list_metric"]?.count()
+        )
+    }
+
+    @Test
+    fun `string list deserializer should correctly parse string lists`() {
+        val persistedSample = mapOf(
+                "store1#telemetry.invalid_string" to "invalid_string",
+                "store1#telemetry.invalid_bool" to false,
+                "store1#telemetry.null" to null,
+                "store1#telemetry.invalid_int" to -1,
+                "store1#telemetry.valid" to listOf("a", "b", "c")
+        )
+
+        val storageEngine = StringListsStorageEngineImplementation()
+
+        // Create a fake application context that will be used to load our data.
+        val context = Mockito.mock(Context::class.java)
+        val sharedPreferences = Mockito.mock(SharedPreferences::class.java)
+        Mockito.`when`(sharedPreferences.all).thenAnswer { persistedSample }
+        Mockito.`when`(context.getSharedPreferences(
+                ArgumentMatchers.eq(storageEngine::class.java.simpleName),
+                ArgumentMatchers.eq(Context.MODE_PRIVATE)
+        )).thenReturn(sharedPreferences)
+
+        storageEngine.applicationContext = context
+        val snapshot = storageEngine.getSnapshot(storeName = "store1", clearStore = true)
+        assertEquals(1, snapshot!!.size)
+        assertEquals(listOf("a", "b", "c"), snapshot["telemetry.valid"])
+    }
+
+    @Test
+    fun `test JSON output`() {
+        val storeNames = listOf("store1", "store2")
+
+        val metric = StringListMetricType(
+            disabled = false,
+            category = "telemetry",
+            lifetime = Lifetime.Ping,
+            name = "string_list_metric",
+            sendInPings = storeNames
+        )
+
+        val list: MutableList<String> = mutableListOf()
```

Comment:
nit: let's change this line to `val list = listOf...` like the other requested change, above.